src/cmd/src/datanode.rs

```diff
@@ -93,31 +93,20 @@ impl TryFrom<StartCommand> for DatanodeOptions {
             opts.mysql_addr = addr;
         }
 
-        match (cmd.metasrv_addr, cmd.node_id) {
-            (Some(meta_addr), Some(node_id)) => {
-                // Running mode is only set to Distributed when
-                // both metasrv addr and node id are set in
-                // commandline options
-                opts.meta_client_opts.metasrv_addr = meta_addr.clone();
-                opts.node_id = node_id;
-                opts.metasrv_addr = Some(vec![meta_addr]);
-                opts.mode = Mode::Distributed;
-            }
-            (None, None) => {
-                opts.mode = Mode::Standalone;
-            }
-            (None, Some(_)) => {
-                return MissingConfigSnafu {
-                    msg: "Missing metasrv address option",
-                }
-                .fail();
-            }
-            (Some(_), None) => {
-                return MissingConfigSnafu {
-                    msg: "Missing node id option",
-                }
-                .fail();
+        if let Some(node_id) = cmd.node_id {
+            opts.node_id = Some(node_id);
+        }
+
+        if let Some(meta_addr) = cmd.metasrv_addr {
+            opts.meta_client_opts.metasrv_addr = meta_addr;
```

Comment:
should we set `opts.metasrv_addr`?

Reply:
`DatanodeOptions:: metasrv_addr ` is already removed since it's redundant.